lite2/client.go

```diff
@@ -944,16 +954,19 @@ func (c *Client) removeWitness(idx int) {
 	}
 }
 
-func (c *Client) removeNoLongerTrustedHeadersRoutine() {
+func (c *Client) pruneRoutine() {
 	defer c.routinesWaitGroup.Done()
 
-	ticker := time.NewTicker(c.removeNoLongerTrustedHeadersPeriod)
+	ticker := time.NewTicker(c.updatePeriod)
```

Comment:
maybe instead of the routine we can use some kind of a counter in VerifyHeader, which we'll use to detect when we need to call Prune

Reply:
Yes I would agree with this implementation. Should I rewrite it then?

Reply:
Yes. It would've been better to do it in a separate PR (leaving only expiration checks here), but I guess it's fine

Reply:
I could create a separate issue and link it to this PR so we better evidence of what this PR did?

Reply:
Please do so 🙏